### ISSUE TITLE
confused about the directions of axes in `heatmap` and `imshow`

### ISSUE THREAD

**Issue by heliosdrm** [jheinen/GR.jl#241]
The [documentation about `heatmap`](https://gr-framework.org/julia-jlgr.html#heatmaps) says:

> The array is drawn with its first value in the upper left corner, so in some cases it may be neccessary to flip the columns (see the example below).

However in that example, the upper left corner represents the most negative value, which is not the first one of the array - but the last one of the first column. It is as if the y-axis were flipped, although it is not.

It also says:

> By default the function will use the row and column indices for the x- and y-axes,

But again in that example, row indices (from 1 to 20) are in the y-axis, and columns (from 1 to 40) in the x-axis.

On the other hand, those two conditions do seem to be the case of the image displayed by `imshow` (where the documentation says nothing about that).

Am I misunderstanding the docs?

**Comment by jheinen** [comment]
The graphics result is the same as in `Plots/pyplot` - it seems we have to clearify the documentation here.

**Comment by heliosdrm** [comment]
#242 might fix the confusion with `heatmap`.

Regarding `imshow`, looks as if it actually printed the transpose of the image represented by the matrix. See:

```julia
julia> square = [0 0 0; 1 0 0]
2×3 Array{Int64,2}:
 0  0  0
 1  0  0

julia> GR.imshow(square)
```
![square](https://user-images.githubusercontent.com/10651028/64485236-682ad480-d21e-11e9-9b8f-cfa5140171fd.png)

**Comment by heliosdrm** [comment]
Yet another detail: #242 makes the docstring of `heatmap` consistent with what it does. But there is also an inconsistency between the interfaces of `heatmap` and `countour`, `contourf` and others.

Those functions also accept a matrix as Z values, such that X values are associated to rows and Y values to columns. Their documentation is coherent with this, and coincided with what it was said of `heatmap` (although `heatmap` actually did the opposite).

Should all those functions be consistent in their treatment of rows and columns wrt. X and Y axes?

**Comment by heliosdrm** [comment]
Already solved in GR 0.47.0
